### PR TITLE
Refactor invoice/invoice line error mapping

### LINKED ISSUE #795: Refactor invoice/invoice line error mapping

### PR DESCRIPTION
**Pull request by roxy-dao**
Closes #795 

- Remove From trait from invoice/invoice line mapping

### CODE REVIEW

File: server/service/src/invoice/inbound_shipment/delete/validate.rs [hunk 1]
```diff
@@ -12,43 +11,26 @@ pub fn validate(
...
-    fn from(_: InvoiceDoesNotExist) -> Self {
-        DeleteInboundShipmentError::InvoiceDoesNotExist
-    }
-}
-
 impl From<InvoiceLinesExist> for DeleteInboundShipmentError {
     fn from(error: InvoiceLinesExist) -> Self {
         DeleteInboundShipmentError::InvoiceLinesExists(error.0)
     }
 }
```
Comment (clemens-msupply) [inline server/service/src/invoice/inbound_shipment/delete/validate.rs:36]: is this one still needed?

File: server/service/src/invoice/validate.rs [hunk 2]
```diff
@@ -96,20 +85,6 @@ pub fn check_invoice_status(
...
-
-    match result {
-        Ok(invoice_row) => Ok(invoice_row),
-        Err(RepositoryError::NotFound) => Err(WithDBError::err(InvoiceDoesNotExist)),
-        Err(error) => Err(WithDBError::db(error)),
-    }
-}
-
-// TODO replace check_invoice_exists with this
 pub fn check_invoice_exists_option(
```
Comment (clemens-msupply) [inline server/service/src/invoice/validate.rs:88]: Could be renamed to `check_invoice_exists` now...

File: server/service/src/invoice_line/validate.rs [hunk 3]
```diff
@@ -33,59 +16,22 @@ pub fn check_line_does_not_exists_new(
...
-    let item_result = ItemRowRepository::new(connection).find_one_by_id(item_id)?;
-
-    match item_result {
-        Some(item) => Ok(item),
-        None => Err(WithDBError::err(ItemNotFound {})),
-    }
-}
-
-// TODO use this one instead of check_item
 pub fn check_item_exists_option(
```
Comment (clemens-msupply) [inline server/service/src/invoice_line/validate.rs:28]: Remove `_option`?

**Review by clemens-msupply — APPROVED** [review]
Thanks this looks like a lot of work! There where existing error tests, right?

File: server/service/src/invoice/inbound_shipment/delete/validate.rs (hunk 1 above)
Comment (roxy-dao) [reply server/service/src/invoice/inbound_shipment/delete/validate.rs:36]: Well, you can delete invoices even if it has lines... so maybe not? 🤔

Comment (clemens-msupply) [reply server/service/src/invoice/inbound_shipment/delete/validate.rs:36]: sorry meant why was the From still needed? the other has been remove...

Comment (roxy-dao) [reply server/service/src/invoice/inbound_shipment/delete/validate.rs:36]: Ah, cause it wasn't being used anywhere and wasn't sure if it was alright to remove now XD

**Pull request merged**